### PR TITLE
Emtfpp for 14 0 0 pre3

### PR DESCRIPTION
**Pull request by omiguelc**
This PR adds commits from EMTF++ that have been merged into the master branch in CMSSW. These commits are requested in the following PR: https://github.com/cms-l1t-offline/cmssw/pull/1228

This also supersedes the PR: https://github.com/cms-l1t-offline/cmssw/pull/1229

### CODE REVIEW

**Comment by omiguelc** [comment]
Howdy, @aloeliger,

I closed the previous PR because there were too many merge conflicts. I created a branch based on the current offline branch y'all have to avoid this issue. I also added a fix for the GMT compilation issue that ocurred in the previous PR. I wasn't able to mount the `data` folder for EMTF++ which is needed to run the emulator though.

**Comment by aloeliger** [comment]
@omiguelc Are the data changes also already present in the main fork of CMSSW?

**Comment by omiguelc** [comment]
Hey @aloeliger,

Yes, the data I tried copying is the same as the files present in the main fork of CMSSW.

File: L1Trigger/L1TMuonEndCapPhase2/data
Comment (aloeliger) [inline L1Trigger/L1TMuonEndCapPhase2/data:1]: @omiguelc Could you just commit this file/these files directly to this PR? I'm not sure any of CMS's checkout facilities will get this correctly.

Comment (omiguelc) [reply L1Trigger/L1TMuonEndCapPhase2/data:1]: Howdy @aloeliger, done! The PR should be ready :)

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no issues with the code checks!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-checks && scram b -k -j 8 code-checks`|

I found no issues with the headers!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 check-headers`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found 2 files that did not meet formatting requirements:
 - L1Trigger/L1TMuon/src/MicroGMTConfiguration.cc
 - L1Trigger/Phase2L1GMT/plugins/Phase2L1TGMTFwdMuonTranslator.cc

Please run `scram b code-format` to auto-apply code formatting
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-format-all`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

This PR passes available unit tests!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b runtests`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation

I found a non-zero return code running the relval workflows for this PR!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`1`|
|command|`eval `scramv1 runtime -sh` && runTheMatrix.py --what upgrade -l 26834.78`|

**Comment by aloeliger** [comment]
Okay. I think this can be merged. I'll do that along with #1228 tomorrow morning.

**Comment by aloeliger** [comment]
@omiguelc This is getting merged. Please open a PR to CMSSW as soon as possible with these commits.

**Comment by omiguelc** [comment]
I don't believe these commits are needed in CMSSW, because they are already there.

**Comment by aloeliger** [comment]
@omiguelc Force of habit. Don't worry about it.

**Comment by aloeliger** [comment]
@omiguelc For our book keeping, which PR includes these commits in CMSSW?

**Comment by omiguelc** [comment]
Howdy @aloeliger,

It's the PR that merged EMTF++ into the main CMSSW branch: https://github.com/cms-sw/cmssw/pull/43766
The commits can be found here:
- https://github.com/cms-sw/cmssw/pull/43766/commits/c955ecdd1bc85f9e0cc7f44c3b74ca844e49cd3f
- https://github.com/cms-sw/cmssw/pull/43766/commits/685390350a7ab11bda1080dbf6a17e1814cddb68
- https://github.com/cms-sw/cmssw/pull/43766/commits/0245ea2f0dc5602399baf5b29bb61c4afb62dff4

The reason there missing from the integration branch is because the EMTF++ package got merged before I applied these changes.

**Pull request merged**